src/resources/material.js

```diff
@@ -238,7 +238,14 @@ class MaterialHandler {
 
             // data[name] contains an asset id for a texture
             // if we have an asset id and nothing is assigned to the texture resource or the placeholder texture is assigned
-            if (data[name] && (!materialAsset.resource[name] || materialAsset.resource[name] === this._getPlaceholderTexture(name, materialAsset))) {
+            // or the data has changed
+            const dataAssetId = data[name];
+
+            const materialTexture = material[name];
+            const dataValidated = data.validated;
+            const isPlaceHolderTexture = materialTexture === this._getPlaceholderTexture(name, materialAsset);
```

Comment:
I'm being a little bit pedantic here, but would suggest placing the two lines that deal with `materialTexture` together.
```suggestion
            const isPlaceHolderTexture = materialTexture === this._getPlaceholderTexture(name, materialAsset);
            const dataValidated = data.validated;
```